core/src/test/java/org/elasticsearch/action/admin/cluster/node/stats/NodeStatsTests.java

```diff
@@ -0,0 +1,371 @@
+                if (nodeStats.getOs() == null) {
+                    assertNull(deserializedNodeStats.getOs());
+                } else {
+                    assertEquals(nodeStats.getOs().getTimestamp(), deserializedNodeStats.getOs().getTimestamp());
+                    assertEquals(nodeStats.getOs().getSwap().getFree(), deserializedNodeStats.getOs().getSwap().getFree());
+                    assertEquals(nodeStats.getOs().getSwap().getTotal(), deserializedNodeStats.getOs().getSwap().getTotal());
+                    assertEquals(nodeStats.getOs().getSwap().getUsed(), deserializedNodeStats.getOs().getSwap().getUsed());
+                    assertEquals(nodeStats.getOs().getMem().getFree(), deserializedNodeStats.getOs().getMem().getFree());
+                    assertEquals(nodeStats.getOs().getMem().getTotal(), deserializedNodeStats.getOs().getMem().getTotal());
+                    assertEquals(nodeStats.getOs().getMem().getUsed(), deserializedNodeStats.getOs().getMem().getUsed());
+                    assertEquals(nodeStats.getOs().getMem().getFreePercent(), deserializedNodeStats.getOs().getMem().getFreePercent());
+                    assertEquals(nodeStats.getOs().getMem().getUsedPercent(), deserializedNodeStats.getOs().getMem().getUsedPercent());
+                    assertEquals(nodeStats.getOs().getCpu().getPercent(), deserializedNodeStats.getOs().getCpu().getPercent());
+                    assertArrayEquals(nodeStats.getOs().getCpu().getLoadAverage(),
+                            deserializedNodeStats.getOs().getCpu().getLoadAverage(), 0);
+                }
+                if (nodeStats.getProcess() == null) {
+                    assertNull(deserializedNodeStats.getProcess());
+                } else {
+                    assertEquals(nodeStats.getProcess().getTimestamp(), deserializedNodeStats.getProcess().getTimestamp());
+                    assertEquals(nodeStats.getProcess().getCpu().getTotal(), deserializedNodeStats.getProcess().getCpu().getTotal());
+                    assertEquals(nodeStats.getProcess().getCpu().getPercent(), deserializedNodeStats.getProcess().getCpu().getPercent());
+                    assertEquals(nodeStats.getProcess().getMem().getTotalVirtual(),
+                            deserializedNodeStats.getProcess().getMem().getTotalVirtual());
+                    assertEquals(nodeStats.getProcess().getMaxFileDescriptors(),
+                            deserializedNodeStats.getProcess().getMaxFileDescriptors());
+                    assertEquals(nodeStats.getProcess().getOpenFileDescriptors(),
+                            deserializedNodeStats.getProcess().getOpenFileDescriptors());
+                }
+                if (nodeStats.getJvm() == null) {
+                    assertNull(deserializedNodeStats.getJvm());
+                } else {
+                    assertEquals(nodeStats.getJvm().getTimestamp(), deserializedNodeStats.getJvm().getTimestamp());
```

Comment:
I feel like it'd be a bit easier to read/less line-wrappy if you made variables for jvm and mem, maybe gc too?